ag-master/okpy.py

```diff
@@ -1,94 +1,105 @@
+                access_token=access_token,
             )
-            for id, job_id in zip(subms, jobs)
+            for backup_id, job_secret in zip(subm_ids, job_secrets)
         ]
-        db.session.bulk_save_objects(objects)
+        db.session.bulk_save_objects(jobs)
         db.session.commit()
 
         trigger_jobs(
-            secret=get_secret(secret_name="AG_MASTER_SECRET"),
-            assignment_id=assignment_id,
-            subms=subms,
-            jobs=jobs,
-            noreply=True,
+            assignment_id=assignment.assignment_secret, jobs=job_secrets, noreply=True
         )
 
-        return dict(jobs=jobs)
+        return dict(jobs=[job.external_job_id for job in jobs])
 
     @trigger_jobs.bind(app)
-    def trigger_jobs_rpc(secret, assignment_id, subms, jobs):
-        if secret != get_secret(secret_name="AG_MASTER_SECRET"):
-            raise PermissionError
-        assignment = Assignment.query.filter_by(ag_key=assignment_id).first()
-
-        subm_batches = [
-            subms[i : i + BATCH_SIZE] for i in range(0, len(subms), BATCH_SIZE)
-        ]
+    @only("ag-master")
+    def trigger_jobs_impl(assignment_id, jobs):
         job_batches = [
             jobs[i : i + BATCH_SIZE] for i in range(0, len(jobs), BATCH_SIZE)
         ]
-
-        for subm_batch, job_batch in zip(subm_batches, job_batches):
-            batch_grade(
-                assignment_id=assignment.ag_key,
-                assignment_name=assignment.name,
-                command=assignment.command,
-                backups=subm_batch,
-                jobs=job_batch,
-                course_key=assignment.course,
-                secret=get_secret(secret_name="AG_WORKER_SECRET"),
-                noreply=True,
-                timeout=8,
-            )
-        return dict(success=True)
+        assignment: Assignment = Assignment.query.get(assignment_id)
+
+        bucket = storage.Client().get_bucket(BUCKET)
+        blob = bucket.blob(f"zips/{assignment.endpoint}/{assignment.file}")
+        with tempfile.NamedTemporaryFile() as temp:
+            blob.download_to_filename(temp.name)
+            with open(temp.name, "rb") as zf:
+                encoded_zip = base64.b64encode(zf.read()).decode("ascii")
+
+        for job_batch in job_batches:
+            try:
+                batch_grade(
+                    command=assignment.command,
+                    jobs=job_batch,
+                    grading_zip=encoded_zip,
+                    noreply=True,
+                    timeout=8,
+                )
+            except:
+                # @nocommit this is somehow wrong because it errored in a PR build
+                Job.query.filter(Job.job_key.in_(job_batch)).update(
+                    {
+                        Job.status: "failed",
+                        Job.result: "trigger_job error\n" + traceback.format_exc(),
+                    }
+                )
+                db.session.commit()
```

Comment:
noreply doesn't throw an exception if this particular request is lost, so this doesn't really catch a lot of trigger errors